--- a/makefile	2023-04-29 13:54:52.570414011 +0300
+++ b/makefile	2023-04-29 13:53:53.889457399 +0300
@@ -108,7 +108,7 @@
 	BUILD_TYPE = Debug
 endif
 
-PUBLISHFLAGS ?=  -r $(MESENPLATFORM) --no-self-contained true -p:PublishSingleFile=true
+PUBLISHFLAGS ?=  -r $(MESENPLATFORM) --self-contained true -p:PublishSingleFile=true
 
 CORESRC := $(shell find Core -name '*.cpp')
 COREOBJ := $(CORESRC:.cpp=.o)
@@ -137,7 +137,7 @@
 	LIBEVDEVINC := -I../
 endif
 
-FSLIB := -lstdc++fs
+FSLIB :=
 
 ifeq ($(MESENOS),osx)
 	LIBEVDEVOBJ := 
@@ -154,8 +154,8 @@
 	rm -fr $(OUTFOLDER)/Dependencies/*
 	cp InteropDLL/$(OBJFOLDER)/$(SHAREDLIB) $(OUTFOLDER)/$(SHAREDLIB)
 	#Called twice because the first call copies native libraries to the bin folder which need to be included in Dependencies.zip
-	cd UI && dotnet publish -c $(BUILD_TYPE) -p:OptimizeUi="true" $(PUBLISHFLAGS)
-	cd UI && dotnet publish -c $(BUILD_TYPE) -p:OptimizeUi="true" $(PUBLISHFLAGS)
+	cd UI && dotnet publish -c $(BUILD_TYPE) --source ../nuget-sources -p:OptimizeUi="true" $(PUBLISHFLAGS)
+	cd UI && dotnet publish -c $(BUILD_TYPE) --source ../nuget-sources -p:OptimizeUi="true" $(PUBLISHFLAGS)
 
 core: InteropDLL/$(OBJFOLDER)/$(SHAREDLIB)
 
